stock_account_change_product_valuation/models/product.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# © 2016 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openerp import api, models
+from openerp.tools.float_utils import float_round
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    @api.multi
+    def write(self, values):
+        if values.get('type', False):
+            new_type = values.get('type')
+            if new_type == 'product':
+                for rec in self:
```

Comment:
@mreficent I would have done this : `for variant in self.filtered(lambda r: r.type == 'consu').mapped('product_variant_ids'):`

That avoid a supplementary loop. Idem for next ones.

Then, I would have modified all quants of variant_ids at once, to have just one write().

So, 
`
variants = self.filtered(lambda r: r.type == 'consu').mapped('product_variant_ids')
quants = self.search([('product_id', 'in', variants.ids)])
quants.write({'cost': 0.0'})
variants.mapped('product_tmpl_id').standard_price = 0.0
`
I'm not sure about the last line. The point is making quite loops in a write is bulky. But that's my point of view ;-)